### PR TITLE
fix(ISubscription): update type definition of ISubscription::closed

### PR DESCRIPTION
**Pull request by kwonoj**
**Description:**

Originated from https://github.com/ReactiveX/rxjs/pull/2234, doesn't add any code changes but only update type definition for interface clarity. 

Currently it's not so benefitable even for TS user since all of codebase just return `Subscription` everywhere instead of using interfaces, that need to be dealt with separate refactoring.

**Related issue (if exists):**
#2234

### CODE REVIEW

File: src/Subscription.ts [hunk 1]
```diff
@@ -13,7 +13,7 @@ export type TeardownLogic = AnonymousSubscription | Function | void;
 
 export interface ISubscription extends AnonymousSubscription {
   unsubscribe(): void;
-  closed: boolean;
+  readonly closed: boolean;
```
Comment (david-driscoll) [inline src/Subscription.ts]: 👍 this will encourage TypeScript users to not try to modify this value (as it shouldn't be mutable)

Comment (kwonoj) [inline src/Subscription.ts]: @david-driscoll check #2234 as well, this PR is subset of it only updated type definition, while it does updates actual code behaviors.

**Comment by blesh** [comment]
> since all of codebase just return Subscription everywhere instead of using interfaces, that need to be dealt with separate refactoring.

Technically, a LOT of the codebase returns subclasses of `Subscriber`. So people *could* call `next` on them if they chose too.. however, that's not supported nor is it documented.

I guess what I'm saying is: Hold off on that refactor, there's likely one coming anyhow... especially with the planned scheduling changes.

**Comment by jayphelps** [comment]
LGTM

**Comment by kwonoj** [comment]
LGTM

**Pull request merged**